pytorch_lightning/trainer/callback_hook.py

```diff
@@ -87,20 +89,38 @@ def on_validation_epoch_start(self):
         for callback in self.callbacks:
             callback.on_validation_epoch_start(self, self.lightning_module)
 
-    def on_validation_epoch_end(self):
+    def on_validation_epoch_end(self, outputs):
         """Called when the epoch ends."""
         for callback in self.callbacks:
-            callback.on_validation_epoch_end(self, self.lightning_module)
+            params = list(inspect.signature(callback.on_validation_epoch_end).parameters)
+            if "outputs" in params:
+                callback.on_validation_epoch_end(self, self.get_model(), outputs)
+            else:
+                rank_zero_warn(
+                    "`Callback.on_validation_epoch_end` signature has changed in v1.3."
+                    "`outputs` parameter has been added."
+                    " Support for the old signature will be removed in v1.5", DeprecationWarning
+                )
```

Comment:
```suggestion
                rank_zero_warn(
                    "`Callback.on_validation_epoch_end` signature has changed in v1.3."
                    " `outputs` parameter has been added."
                    " Support for the old signature will be removed in v1.5", DeprecationWarning
                )
```

Reply:
quick question: will this warning be logged after each val loop?
also add a test for it in deprecated_api?

Reply:
Yes, every time `on_evaluation_epoch_end` is called.

Reply:
will it be good to log them only once somehow?

Reply:
Yeah, would have to store an attribute or a property in the Class itself.

Reply:
You can use the warning cache

Reply:
Nicee! Updated